render/container.go

```diff
@@ -71,35 +68,28 @@ func (c connectionJoin) Render(rpt report.Report) Nodes {
 			}
 		}
 	}
-	ret := newJoinResults(inputNodes.Nodes)
-
-	// Now look at all the endpoints and see which map to IP nodes
-	for _, m := range endpoints.Nodes {
-		scope, addr, port, ok := report.ParseEndpointNodeID(m.ID)
-		if !ok {
-			continue
-		}
-		// Nodes without a hostid may be pseudo nodes - if so, pass through to result
-		if _, ok := m.Latest.Lookup(report.HostNodeID); !ok {
-			if id, ok := externalNodeID(m, addr, local); ok {
-				ret.addChild(m, id, Pseudo)
-				continue
+	return MapEndpoints(
+		func(m report.Node) string {
+			scope, addr, port, ok := report.ParseEndpointNodeID(m.ID)
+			if !ok {
+				return ""
 			}
-		}
-		id, found := ipNodes[report.MakeScopedEndpointNodeID(scope, addr, "")]
-		// We also allow for joining on ip:port pairs.  This is useful for
-		// connections to the host IPs which have been port mapped to a
-		// container can only be unambiguously identified with the port.
-		if !found {
-			id, found = ipNodes[report.MakeScopedEndpointNodeID(scope, addr, port)]
-		}
-		if found && id != "" { // not one we blanked out earlier
-			// We are guaranteed to find the id, so really this should
-			// never end up creating a node.
-			ret.addChild(m, id, report.Container)
-		}
-	}
-	return ret.result(endpoints)
+			id, found := ipNodes[report.MakeScopedEndpointNodeID(scope, addr, "")]
+			// We also allow for joining on ip:port pairs.  This is
+			// useful for connections to the host IPs which have been
+			// port mapped to a container can only be unambiguously
+			// identified with the port.
+			if !found {
+				id, found = ipNodes[report.MakeScopedEndpointNodeID(scope, addr, port)]
+			}
+			if !found || id == "" {
+				return ""
+			}
+			// Not an IP we blanked out earlier. We are guaranteed to
+			// find the id, so really this should never end up
+			// creating a node.
+			return id
```

Comment:
It seems this comment has got a bit separated from the code that makes it make sense.
It would be `joinResult.addChild()` that created a node, right?
And the 'find' part is also under there.  How can we know here what will be in `joinResult.nodes` ?

Reply:
We know because the id is the result of a lookup in `ipNodes`, and all the ids in `ipNodes` are ids of nodes in `c.topology`.

Do you reckon I should say that in the comment?